src/Components/Patient/PatientConsentRecords.tsx

```diff
@@ -91,55 +124,36 @@ export default function PatientConsentRecords(props: {
     };
   };
 
-  const handleUpload = async (diffPCS?: ConsentRecord) => {
-    if (newConsent.type === 0) return;
-    const consentTypeExists = consentRecords?.find(
-      (record) => record.type === newConsent.type && record.deleted !== true,
+  const handleUpload = async (diffPCS?: PatientConsentModel) => {
+    const consentExists = consentRecords?.find(
+      (record) => record.type === newConsent.type && !record.archived,
     );
-    if (consentTypeExists && !diffPCS) {
-      await fileUpload.handleFileUpload(consentTypeExists.id);
-    } else {
-      const randomId = "consent-" + new Date().getTime().toString();
-      const newRecords = [
-        ...(consentRecords?.map((r) =>
-          r.id === diffPCS?.id ? { ...r, deleted: true } : r,
-        ) || []),
-        {
-          id: randomId,
-          type: newConsent.type,
+    let consentId = consentExists?.id;
+    if (!consentExists || diffPCS) {
+      consentId = undefined;
+      const res = await request(routes.createConsent, {
+        pathParams: { consultationId: consultationId },
+        body: {
+          ...newConsent,
           patient_code_status:
             newConsent.type === 2 ? newConsent.patient_code_status : undefined,
         },
-      ];
-      await request(routes.partialUpdateConsultation, {
-        pathParams: { id: consultationId },
-        body: { consent_records: newRecords },
       });
-      await fileUpload.handleFileUpload(randomId);
-      setConsentRecords(newRecords);
+      if (res.data) {
+        consentId = res.data.id;
+      }
     }
-
+    consentId && (await fileUpload.handleFileUpload(consentId));
     refetch();
```

Comment:
you could technically avoid the hacks and call the refetch after the `handleUpload` is completed directly inside the PaginatedList component's child on the button's onClick

Reply:
<img width="914" alt="image" src="https://github.com/coronasafe/care_fe/assets/25143503/2a882e17-237a-43d6-b5d7-1b98bbba12c4">

<img width="1023" alt="image" src="https://github.com/coronasafe/care_fe/assets/25143503/5f22989e-8a13-4edb-a205-68bc7a35a301">